qemu/tests/blockdev_stream_filter_nodename.py

```diff
@@ -0,0 +1,60 @@
+from provider import blockdev_stream_nowait
+from provider import job_utils
+
+
+class BlkdevStreamFilterNode(blockdev_stream_nowait.BlockdevStreamNowaitTest):
+    """
+    blockdev-stream filter node name test
+    """
+
+    def _init_stream_options(self):
+        super(BlkdevStreamFilterNode, self)._init_stream_options()
+        filter_node_name = self.params.get("filter_node_name")
+        if filter_node_name:
+            self._stream_options["filter-node-name"] = filter_node_name
+
+    def check_filter_nodes_name(self):
+        """
+        Check filter node name set during stream
+        """
+        blocks_info = self.main_vm.monitor.query("block")
+        for block in blocks_info:
+            block_node_name = block["inserted"].get("node-name")
+            if (self.params.get("source_images") in block["qdev"] and
+                    block_node_name == self.params['filter_node_name']):
+                break
+        else:
+            self.test.fail("Filter node name '%s' is not set as expected"
+                           "during stream" % self.params['filter_node_name'])
+
+    def do_test(self):
+        self.snapshot_test()
+        self.blockdev_stream()
+        job_utils.check_block_jobs_started(
+            self.main_vm, [self._job],
+            self.params.get_numeric('job_started_timeout', 60)
+        )
+        self.check_filter_nodes_name()
+        self.wait_stream_job_completed()
+
```

Comment:
Better to check block info again to make sure there is no 'filter_node_name' after the stream job finished.

Reply:
Have updated, please help to review again, thanks